src/cdk/tree/tree.ts

```diff
@@ -165,15 +172,34 @@ export class CdkTree<T> implements CollectionViewer, OnInit, OnDestroy {
     }
 
     this._dataSource = dataSource;
+    if (this._nodeDefs) {
+      this._observeRenderChanges();
+    }
   }
 
   /** Set up a subscription for the data provided by the data source. */
   private _observeRenderChanges() {
-    this._dataSubscription = this.dataSource.connect(this).pipe(takeUntil(this._onDestroy))
-      .subscribe(data => {
-        this._data = data;
-        this._renderNodeChanges(data);
-      });
+    let dataStream: Observable<T[]> | undefined;
+
+    // Cannot use `instanceof DataSource` since the data source could be a literal with
+    // `connect` function and may not extends DataSource.
+    if ((this._dataSource as DataSource<T>).connect instanceof Function) {
```

Comment:
I missed this in the table PR, but it should use `typeof` rather than `instanceof` since the former is generally more reliable for primitive types. 

Also it should use the same checks for the `instanceof DataSource` checks in the rest of the file (which I also missed in the table PR)

Reply:
Updated. Thanks for review!